pkg/ccn-lite/patches/0004-src-ccnl-core-conditional-definition-of-gettimeofday.patch

```diff
@@ -0,0 +1,33 @@
+From 29e12d4f39452c6aaedef6c99f94ac93f59df7d2 Mon Sep 17 00:00:00 2001
+From: Gunar Schorcht <gunar@schorcht.net>
+Date: Fri, 12 Aug 2022 15:23:59 +0200
+Subject: [PATCH 4/4] src/ccnl-core: conditional definition of gettimeofday
+
+The definition of gettimeofday in this file leads to multiple definitions on platforms where gettimeofday is compiled in in newlib. The definition has to be conditional dependent on the MCU.
+---
+ src/ccnl-core/src/ccnl-os-time.c | 2 ++
+ 1 file changed, 2 insertions(+)
+
+diff --git a/src/ccnl-core/src/ccnl-os-time.c b/src/ccnl-core/src/ccnl-os-time.c
+index 0087b9dc156..986afa5008e 100644
+--- a/src/ccnl-core/src/ccnl-os-time.c
++++ b/src/ccnl-core/src/ccnl-os-time.c
+@@ -38,6 +38,7 @@ struct ccnl_timer_s *eventqueue;
+ 
+ 
+ #if defined(CCNL_RIOT) && !(defined(__FreeBSD__) || defined(__APPLE__) || defined(__linux__))
++#if !defined(MCU_ESP32)
+ #include <ztimer64.h>
+ #include <timex.h>
+ #include <div.h>
+@@ -51,6 +52,7 @@ int gettimeofday(struct timeval *__restrict __p, void *__restrict __tz)
```

Comment:
Would this work?

```C
__attribute__((weak))
int gettimeofday(struct timeval *__restrict __p, void *__restrict __tz)
```

That would be easier to get upstream

Reply:
I removed the patch and added a patch which declares `gettimeofday` in `ccn-lite` package as weak symbol.